docs/src/getting_started/using_cmaize_as_build_system/cmake_like_project.rst

```diff
@@ -0,0 +1,174 @@
+is common to CMake projects that uses CMaize as its build system.
+
+Project Overview
+================
+
+``MyCMakeProject`` is a simple C++ project that consists of the following
+components:
+
+* The ``TravelCostUtils`` library that contains some simple arithmetic
+  calculations for use in calculating travel costs.
+
+* The ``TravelCostCalc`` executable that calculates travel cost for a set of
+  predefined values using functions in ``TravelCostUtils``.
+
+* The ``Test`` directory contains some tests written using the Catch2 library
+  that test that ``TravelCostUtils`` works properly.
+
+Here is the file structure of the project:
+
+::
+
+    MyCMakeProject
+    ├── CMakeLists.txt
+    ├── cmake
+    │   └── get_cpp.cmake
+    ├── TravelCostUtils
+    |   ├── TravelCostUtils.cpp
+    |   └── TravelCostUtils.hpp
+    ├── TravelCostCalc
+    |   └── TravelCostCalc.cpp
+    └── Test
+        └── test.cpp
+
+This structure is commonly found in CMake projects. A subdirectory is created
+for each executable and library. A subdirectory is also created the project's
+tests.
+
+Creating the CMakeLists.txt File
+================================
+
+Here we will cover the contents of the ``CMakeLists.txt`` file at the root of
+the project.
+
+The Basics
+^^^^^^^^^^
+
+First we must start with the basics. We need to declare the minimum required
+version of CMake, the name and version of the project, and the C++ standard
+of the project.
+
+.. code-block:: cmake
+
+    # Declare minimum version required
+    cmake_minimum_required(VERSION 3.13.4)
+
+    # Declare project name and version
+    project(MyCMakeProject VERSION 1.0)
+
+    # Set C++ standard
+    set(CMAKE_CXX_STANDARD 14)
+
+Including CMaize
+^^^^^^^^^^^^^^^^
+
+We also need to include CMaize. This is described more in
+:ref:`Automatically Downloading and Including CMaize`.
+
+.. code-block:: cmake
+
+    # Include CMaize automatically
+    include("${PROJECT_SOURCE_DIR}/cmake/get_cpp.cmake")
+
+Adding the TravelCostUtils Library
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now we will add our ``TravelCostUtils`` library. We need to specify the target
+name, its source directory, and its include directory:
+
+.. code-block:: cmake
+
+    # Add our TravelCostUtils library
+    cpp_add_library(
+        TravelCostUtils
+        SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/TravelCostUtils"
+        INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/TravelCostUtils"
+    )
+
+Adding the TravelCostCalc Executable
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here we add our ``TravelCostCalc`` executable. We simply need to specify
+the target name, its source directory, and its dependencies:
+
+.. code-block:: cmake
+
+    # Add the executable
+    cpp_add_executable(
+        TravelCostCalc
+        SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/TravelCostCalc"
+        DEPENDS TravelCostUtils
+    )
+
+Getting the Catch2 Unit Test Framework
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```

Comment:
The getting of Catch2 and the building of the unit tests should be inside an `if("${BUILD_TESTING}")...endif()` block. `BUILD_TESTING` is the official CMake variable that controls whether or not unit testing is done. If the user is not building unit tests then there's no point in looking for Catch2.

Reply:
I fixed this in both the docs and example projects.